### PR TITLE
Dbthread join recover deadlock

### PR DESCRIPTION
**Pull request by markhannum**
recover_deadlock in dbthread_join

### CODE REVIEW

File: lua/sp.c [hunk 1]
```diff
@@ -3227,6 +3227,14 @@ static int dbthread_join(Lua lua1)
         struct timespec ts;
         clock_gettime(CLOCK_REALTIME, &ts);
         ts.tv_sec += 1;
+        if (bdb_lock_desired(thedb->bdb_env)) {
+            int rc;
+            if ((rc = recover_deadlock(thedb->bdb_env, lt->sp->thd->sqlthd, NULL, 0)) !=
+                    0) {
+                luabb_error(lua1, lt->sp, "recover deadlock failed");
+                return -3;
```
Comment (adizaimi) [inline lua/sp.c]: Can we look into why recover_deadlock could fail here, and what is best to do when it does?

Comment (markhannum) [reply lua/sp.c]: Discussed with Akshat- he is going to take care of this ..

Comment (akshatsikarwar) [reply lua/sp.c]: I will "take care" of this.

**Comment by markhannum** [comment]
Handing to Akshat.  He is going to change the behavior to force the lua program to stop with a failure condition if recover_deadlock fails.

**Pull request closed without merging**